File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.cpp
class Solution {
public:
    double findMedian(vector<int>& nums){
        int n = nums.size();
        if(n%2 == 0){
            return double(nums[n/2-1]+nums[n/2])/2;
        }
        return double(nums[n/2]);
    }
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        vector<int>nums;
        int n = nums1.size(), m = nums2.size(), i = 0, j = 0;
        while(i<n && j<m){
            if(nums1[i]>nums2[j])
                nums.push_back(nums2[j++]);
            else if(nums1[i]<nums2[j])
                nums.push_back(nums1[i++]);
            else {
                nums.push_back(nums1[i++]);
                nums.push_back(nums2[j++]);
            }
        }
        while(i<n){
            nums.push_back(nums1[i++]);
        }
        while(j<m){
            nums.push_back(nums2[j++]);
        }
        return findMedian(nums);
    }
};

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int n = nums1.length, m = nums2.length, i=0, j=0, k=0;
        ArrayList<Integer> nums = new ArrayList<Integer>(m+n);
        while(i<n && j<m){
            if(nums1[i] <= nums2[j]){
                nums.add(nums1[i]);
                i++;
            }
            else{
                nums.add(nums2[j]);
                j++;
            }
        }
        while(i<n){
                nums.add(nums1[i]);
                i++;
        }
        while(j<m){
                nums.add(nums2[j]);
                j++;
        }
        if((m+n)%2==1)
            return nums.get((m+n)/2);
        return (nums.get((m+n)/2)+nums.get((m+n)/2-1))/2.0;
    }
}